pallets/xcmp-queue/src/lib.rs

```diff
@@ -141,6 +144,8 @@ pub mod pallet {
 		) -> DispatchResultWithPostInfo {
 			T::ExecuteOverweightOrigin::ensure_origin(origin)?;
 
+			ensure!(!T::MaintenanceStatusProvider::is_maintenance(), Error::<T>::XcmMsgProcessingBlockedByMaintenanceMode);
```

Comment:
should that rather early return instead ?

Reply:
It is returning early, right? Or do you mean even before the ensure_origin check?